packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts

```diff
@@ -60,22 +61,114 @@ export default class MuiDetailGenerator implements DetailGenerator {
         );
 
         if (widgetParentNode) {
-          let formikDeclarationNode = this.findGridDeclaration(widgetParentNode);
+          let formikDeclarationNode =
+            this.findGridDeclaration(widgetParentNode);
 
           if (formikDeclarationNode) {
-            let propertyAssigmentArray: ts.PropertyAssignment[] = []; 
-            findPropertyAssignment(formikDeclarationNode.getChildAt(1), propertyAssigmentArray);
+            let propertyAssigmentArray: ts.PropertyAssignment[] = [];
+            findPropertyAssignment(
+              formikDeclarationNode.getChildAt(1),
+              propertyAssigmentArray
+            );
+
+            if (propertyAssigmentArray) {
+              ast = this.addNewField(
+                formikDeclarationNode,
+                propertyAssigmentArray,
+                property,
+                ast
+              );
+            }
+          }
+        }
+
+        let gridElements: ts.JsxElement[] = [];
+        //Add grid with field
+        if (widgetParentNode) {
+          this.findGridElement(widgetParentNode, gridElements);
+        }
 
-             if (propertyAssigmentArray) {
-               ast = this.addNewField(formikDeclarationNode, propertyAssigmentArray, property, ast);
-             }
+        if (gridElements) {
+          if (widgetParentNode) {
+            let gridContainers: ts.JsxElement[] = [];
+            this.findGridContainer(widgetParentNode, gridContainers);
```

Comment:
This is strange C/C++ like pattern for returning value using parameter. Please, change that using "usual" `return`

Reply:
Im not sure how to do it because it is recursive function and i need to fill an array of founded elements.

Reply:
> Im not sure how to do it because it is recursive function and i need to fill an array of founded elements.

const ret1 = recursiveCall()
const toBeAdded = [1, 2, 3]
return [...ret1, toBeAdded]